src/Data/YAML/Loader.hs

```diff
@@ -46,12 +46,12 @@ data Loader m n = Loader
 -- represented as 'Text' values). See also 'decodeNode' for a more
 -- convenient interface.
 {-# INLINEABLE decodeLoader #-}
-decodeLoader :: forall n m . MonadFix m => Loader m n -> BS.L.ByteString -> m (Either String [n])
+decodeLoader :: forall n m . MonadFix m => Loader m n -> BS.L.ByteString -> m (Either (YE.Pos, String) [n])
 decodeLoader Loader{..} bs0 = do
     case sequence . YE.parseEvents $ bs0 of
       Left (pos,err)
-        | YE.posCharOffset pos < 0 -> return (Left err)
-        | otherwise                -> return (Left $ ":" ++ show (YE.posLine pos) ++ ":" ++ show (YE.posColumn pos) ++ ": " ++ err)
+        | YE.posCharOffset pos < 0 -> return (Left (pos,err))
+        | otherwise                -> return (Left (pos,":" ++ show (YE.posLine pos) ++ ":" ++ show (YE.posColumn pos) ++ ": " ++ err))
```

Comment:
btw, we don't need to include the line/column position in the error string anymore